example/lib/main.dart

```diff
@@ -1,53 +1,69 @@
 // ignore_for_file: undefined_class, uri_has_not_been_generated
 import 'package:data_classes/data_classes.dart';
 
+// ignore: unused_import
+import 'models_helpers.dart';
+export 'models_helpers.dart';
+
 part 'main.g.dart';
 
-void main() {
-  final apple = Fruit(
-    color: Color.green,
-    name: 'apple',
-    shape: Shape.round,
+const pineappleJson = {
+  'color': 'yellow',
+  'extraInfo': {
+    'cultivatedIn': 'Costa Rica',
+    'priceInDollarsPerKg': 3.2,
+  },
+  'name': 'Pineapple',
+  'weightInGrams': 500,
+};
+
+Future<void> main() async {
+  final pineapple = Fruit.fromJson(pineappleJson);
+  print('Pineapple: $pineapple');
+
+  final heavyPineapple = pineapple.copyWith(weight: 1500);
+  print('Heavy pineapple: $heavyPineapple');
+
+  final pineappleFromBrasil = await pineapple.copyAsync(
+    (builder) async => builder.extraInfo = {
+      ...builder.extraInfo,
+      ...await fetchPineappleInfo(),
+    },
   );
-  final orange1 = Fruit.fromJson(orangeJson);
-  final orange2 = Fruit.fromJson(orangeJson);
-  final yellowFruit = Fruit(color: Color.yellow);
-  final lime = yellowFruit.copy((f) => f.name = 'lime');
-  apple.big = true; // should be analyzer error
-
-  print('Apple = $apple\nLime = $lime\nOrange = $orange1');
```

Comment:
@dndoja , can you keep these prints? They help to ensure that the equality and copying works as expected. Or even better let's also create a few simple tests for that.